src/Reflection/ReflectionMethod.php

```diff
@@ -94,8 +94,22 @@ public function getPrototype() : self
                 continue;
             }
 
-            if ($i->hasMethod($this->getName()) && $i->getMethod($this->getName())->isAbstract()) {
-                return $i->getMethod($this->getName());
+            if ($i->hasMethod($this->getName())) {
+                $method = $i->getMethod($this->getName());
+
+                if ($method->isAbstract()) {
```

Comment:
Maybe these conditions should bein a private method that we can call on the other instance? This includes the exception

Reply:
Refactored in new commit https://github.com/Roave/BetterReflection/pull/350/commits/a70d5b5700399759a7a669cc082aaa2fcd970e38

However I'm really not happy with the name `findPrototype()`. I hope someone find better name.